HousingSearchApi.Tests/V2/E2ETests/PersonSearchTests.cs

```diff
@@ -91,35 +92,23 @@ public async Task SearchTenure_PaymentRef()
         // Assert
         response.StatusCode.Should().Be(HttpStatusCode.OK);
         var root = GetResponseRootElement(response);
-        root.GetProperty("total").GetInt32().Should().BeGreaterThan(0);
-        var firstResult = root.GetProperty("results").GetProperty("persons")[0];
-        firstResult.GetProperty("tenures")[0].GetProperty("paymentReference").GetString().Should().Be(searchText);
+        try
+        {
+            root.GetProperty("total").GetInt32().Should().BeGreaterThan(0);
+            var firstResult = root.GetProperty("results").GetProperty("persons")[0];
+            firstResult.GetProperty("tenures")[0].GetProperty("paymentReference").GetString().Trim().Should().Be(searchText.Trim());
+        }
+        catch (AssertionException e)
+        {
+            var errMsg = $"Failed to assert that the payment reference is {searchText}";
+            throw new AssertionException(errMsg + "\n" + e.Message);
+        }
```

Comment:
Gives more useful error logs - added to this test because it was unexpectedly flaky, but should be added to any other flaky tests in future